src/SVDiagnosticCollection.ts

```diff
@@ -182,51 +175,26 @@ export class SVDiagnosticCollection implements Disposable {
      * Also goes through the codeflow locations, to update the locations
      */
     public async mappingChanged(): Promise<void> {
-        for (const key of this.issuesCollection.keys()) {
-            const issues: SarifViewerVsCodeDiagnostic[] | undefined = this.issuesCollection.get(key);
-            if (!issues) {
-                continue;
+        for (const issues of this.issuesCollection.values()) {
+            for (const issue of issues) {
+                await SVDiagnosticFactory.tryToRemapLocations(this.fileMapper, issue);
             }
-
-            for (const index of issues.keys()) {
-                await SVDiagnosticFactory.tryToRemapLocations(this.explorerController, issues[index]);
-            }
-        }
-
-        const explorerDiag: SarifViewerVsCodeDiagnostic | undefined = this.explorerController.activeDiagnostic;
-        if (!explorerDiag) {
-            return;
         }
 
-        for (const key of this.unmappedIssuesCollection.keys()) {
-            const issues: SarifViewerVsCodeDiagnostic[] | undefined = this.unmappedIssuesCollection.get(key);
-            if (!issues) {
-                return;
-            }
-
+        for (const [key, unmappedIssues] of this.unmappedIssuesCollection.entries()) {
             const remainingUnmappedIssues: SarifViewerVsCodeDiagnostic[] = [];
-            for (const index of issues.keys()) {
-                const diag: SarifViewerVsCodeDiagnostic = issues[index];
-                await SVDiagnosticFactory.tryToRemapLocations(this.explorerController, diag).then((remapped) => {
-                    if (remapped) {
-                        this.add(diag);
-                        this.diagnosticCollectionChangedEventEmitter.fire({
-                            diagnostics: [diag],
-                            type: 'Add'
-                        });
-                        if (explorerDiag !== undefined && explorerDiag.resultInfo.runId === diag.resultInfo.runId &&
-                            explorerDiag.resultInfo.id === diag.resultInfo.id) {
-                            this.explorerController.setActiveDiagnostic(diag, true);
-                        }
-                    } else {
-                        remainingUnmappedIssues.push(issues[index]);
-                    }
-                });
+            for (const unmappedIssue of unmappedIssues) {
+                const remapped: boolean = await SVDiagnosticFactory.tryToRemapLocations(this.fileMapper, unmappedIssue);
+                if (remapped) {
+                    this.add(unmappedIssue);
+                } else {
+                    remainingUnmappedIssues.push(unmappedIssue);
+                }
             }
 
             if (remainingUnmappedIssues.length === 0) {
                 this.unmappedIssuesCollection.delete(key);
-            } else if (remainingUnmappedIssues.length !== issues.length) {
+            } else if (remainingUnmappedIssues.length !== unmappedIssues.length) {
```

Comment:
>(remainingUnmappedIssues.length !== unmappedIssues.length)  [](start = 22, length = 59)

This check isn't actually necessary since setting the value is harmless if nothing has changed, and it is actually safer in case for some reason the logic above changes and the items change but the length stays the same.